src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,1411 @@
+              key: sourceInsert.key,
+            },
+            source_type_name: sourceTypeInsert.name,
+            start_time: sourceInsert.start_time,
+            valid_at: sourceInsert.valid_at,
+          };
+          gridRowSizes = gridRowSizesBottomPanel;
+        }
+      }
+    } else {
+      showFailureToast('Upload failed.');
+    }
+    // Reset the form behind the source
+    parsed = undefined;
+    keyField.reset('');
+    sourceTypeField.reset('');
+    startTimeDoyField.reset('');
+    endTimeDoyField.reset('');
+    validAtDoyField.reset('');
+    derivationGroupField.reset('');
+  }
+
+  async function selectSource(detail: ExternalSourceSlim) {
+    selectedSource = detail;
+    gridRowSizes = gridRowSizesBottomPanel;
+    deselectEvent();
+    eventTooltip.reset();
+  }
+
+  function deselectSource() {
+    deselectEvent();
+    eventTooltip.reset();
+    gridRowSizes = gridRowSizesNoBottomPanel;
+    selectedSource = null;
+  }
+
+  function deselectEvent() {
+    selectedEvent = null;
+    selectedRowId = null;
+  }
+
+  function onCanvasMouseDown(e: CustomEvent<MouseDown>) {
+    if (!mouseDownAfterTable) {
+      const { externalEvents } = e.detail;
+      // selectedEvent is our source of an ExternalEvent as well as the ExternalEventId used by this instance
+      //    of the LayerExternalSources (as opposed to using a store, like the timeline one does, which is
+      //    unnecessary as everything we need is in on single component or can be passed down via parameters to
+      //    children).
+      selectedEvent = externalEvents?.length ? externalEvents[0] : null;
+      selectedRowId = null;
+    } else {
+      mouseDownAfterTable = false;
+    }
+  }
+
+  function onCanvasMouseOver(e: CustomEvent<MouseOver>) {
+    mouseOver = e.detail; // Allows tooltip to access object
+  }
+
+  function toggleItem(value: ExternalSourceType) {
+    if (!selectedFilters.find(f => f.name === value.name)) {
+      selectedFilters = selectedFilters.concat(value);
+    } else {
+      selectedFilters = selectedFilters.filter(filter => filter.name !== value.name);
+    }
+  }
+
+  function unselectFilteredValues() {
+    selectedFilters = [];
+  }
+
+  function selectFilteredValues() {
+    selectedFilters = [...new Set([...selectedFilters, ...filteredValues])];
+  }
+
+  function onSelectionChanged() {
+    selectedEvent = selectedEvents.find(event => getRowIdExternalEvent(event.pkey) === selectedRowId) ?? null;
+  }
+
+  function onManageGroupsAndTypes() {
+    effects.manageGroupsAndTypes(user);
+  }
+</script>
+
+<CssGrid columns={uiColumnSize}>
+  <Panel borderRight padBody={true}>
+    <svelte:fragment slot="header">
+      <SectionTitle
+        >{selectedEvent
+          ? `Selected Event (${selectedEvent.pkey.key})`
+          : selectedSource
+            ? `Selected External Source (${selectedSource.pkey.key})`
+            : 'Upload a Source File'}</SectionTitle
+      >
+      {#if selectedEvent}
```

Comment:
Would be good to inline these two buttons and give the "dismiss" one the `secondary` class.
<img width="505" alt="image" src="https://github.com/user-attachments/assets/559d3bd5-2708-4934-a021-ec191491caba">